protocol_rfcs/geospatial-types.md

```diff
@@ -0,0 +1,89 @@
+- **Writers**, if writing statistics, must use the correct [type metadata](#type-definitions) when computing the covering bounding box.
+- **Writers** must write data files using the [geospatial Parquet logical types](#geospatial-data-in-parquet).
+
+### Type definitions
+
+In the schema the geospatial types are serialized as:
+- `geometry(<crs>)`
+- `geography(<crs>, <algorithm>)`
+
+Both types are parameterized by a **Coordinate Reference System (CRS)**. Additionally, the geography type includes an **edge interpolation algorithm**, that accounts for the model of the Earth (spherical, ellipsoidal) that is used to interpolate between vertices of geospatial objects.
+
+#### Coordinate Reference System
+Coordinate reference system defines the reference frame for interpreting geographic data. It ensures that coordinates are understood in the correct context.
+For example, a CRS for North America may have a different origin (0, 0) than a global CRS, affecting how the coordinates are interpreted.
+
+CRS value can be specified in one of the following formats:
+- A standard authority and identifier (`<authority>:<identifier>`), e.g.:
+    - `OGC:CRS84`
+    - `EPSG:3857`
+- A custom definition, which can be provided in one of two ways:
+    - Using a Spatial Reference System Identifier (SRID), e.g. `srid:<number>`.
+    - Using a projjson reference to a table property where the projjson string is stored, e.g. `projjson:<tableProperty>`.
+
+#### Edge interpolation algorithm
+
+Specifies the algorithm used to interpolate edges between consecutive points. The chosen algorithm determines how intermediate points along an edge are interpreted during spatial operations. It can have one of the following values:
+
+* `spherical`: edges are interpolated as geodesics on a sphere.
+* `vincenty`: [https://en.wikipedia.org/wiki/Vincenty%27s_formulae](https://en.wikipedia.org/wiki/Vincenty%27s_formulae)
+* `thomas`: Thomas, Paul D. Spheroidal geodesics, reference systems, & local geometry. US Naval Oceanographic Office, 1970.
+* `andoyer`: Thomas, Paul D. Mathematical models for navigation systems. US Naval Oceanographic Office, 1965.
+* `karney`: [Karney, Charles FF. "Algorithms for geodesics." Journal of Geodesy 87 (2013): 43-55](https://link.springer.com/content/pdf/10.1007/s00190-012-0578-z.pdf), and [GeographicLib](https://geographiclib.sourceforge.io/)
+
+### Per-file statistics
+
+For geospatial types, the most useful statistics for data skipping are **bounding boxes**. A bounding box is a minimal axis-aligned rectangle, parallelepiped, or hyper-parallelepiped, that fully contains all points of all geometries or geographies in a given set. It is typically represented by two points: the **lower-left** and **upper-right** corners. Each point must include **X** and **Y** coordinates, but may also optionally include **Z** (elevation) and **M** (measure) values.
```

Comment:
>  It is typically represented by two points: the **lower-left** and **upper-right** corners. Each point must include **X** and **Y** coordinates, but may also optionally include **Z** (elevation) and **M** (measure) values.


We need to be more precise here, this doesn't actually specify the format that must be used.
E.g.:
A bound box is represented by two points:
- the **lower-left** corner, defined as the minimum value
- the **upper-right** corner defined as the maximum value.

A point is represented by two floating values, encoded in **[Well-Known Text (WKT)](https://libgeos.org/specifications/wkt/)** format, e.g.:`POINT(-122.419 37.774)`